mpc-core/src/garble/label/mod.rs

```diff
@@ -146,10 +158,8 @@ where
 
     /// Validates whether the provided active labels are authentic
     pub fn validate(&self, labels: &Labels<G, state::Active>) -> Result<(), LabelError> {
-        for (pair, label) in self.state.to_labels().into_iter().zip(&labels.state.labels) {
-            if label.value == pair.low() || label.value == pair.high() {
-                continue;
-            } else {
+        for (pair, label) in self.state.iter().zip(labels.iter()) {
+            if label.value != pair.low() && label.value != pair.high() {
```

Comment:
This is harder to understand as opposed to a simple
` if ! (label.value == pair.low() || label.value == pair.high()) `